v1/contrib/lambda/wrapper.go

```diff
@@ -0,0 +1,367 @@
+	} else if json.Unmarshal(msg, evtv1) == nil {
+		method = evtv1.HTTPMethod
+		headers = evtv1.Headers
+		args = append(args, "HTTPMethod", method)
+		args = append(args, "URL", evtv1.Path)
+		args = append(args, "APIGatewayResourceID", evtv1.RequestContext.ResourceID)
+		args = append(args, "APIGatewayStage", evtv1.RequestContext.Stage)
+	} else if json.Unmarshal(msg, evt) == nil {
+		method = evt.HTTPMethod
+		headers = evt.Headers
+		args = append(args, "Method", method)
+	}
+
+	if headers != nil {
+		for k, v := range headers {
+			k = strings.ToLower(k)
+			switch k {
+			case AOHTTPHeader:
+				mdStr = v
+			case "host":
+				host = v
+				args = append(args, "HTTP-Host", v)
+			case "x-forwarded-for":
+				args = append(args, "Forwarded-For", v)
+			case "x-forwarded-proto":
+				args = append(args, "Forwarded-Proto", v)
+			case "x-forwarded-port":
+				args = append(args, "Forwarded-Port", v)
+			}
+		}
+	}
+
+	if method != "" && host != "" {
+		args = append(args, "Spec", "aws-lambda:ws")
+	} else {
+		args = append(args, "Spec", "aws-lambda")
+	}
+
+	return mdStr, method, args
+}
+
+func (w *traceWrapper) Before(ctx context.Context, msg json.RawMessage, args ...interface{}) context.Context {
+	mdStr, method, args := w.getRequestContext(ctx, msg, args...)
+
+	lc, ok := lambdacontext.FromContext(ctx)
+	if ok {
+		args = append(args, "AwsRequestID", lc.AwsRequestID)
+		args = append(args, "InvokedFunctionArn", lc.InvokedFunctionArn)
+		args = append(args, "FunctionName", lambdacontext.FunctionName)
+		args = append(args, "FunctionVersion", lambdacontext.FunctionVersion)
+		args = append(args, "AWSRegion", os.Getenv("AWS_REGION"))
+		args = append(args, "MemoryLimitInMB", lambdacontext.MemoryLimitInMB)
+		args = append(args, "LogStreamName", lambdacontext.LogStreamName)
+	}
+
+	cb := func() ao.KVMap {
+		m := ao.KVMap{}
+		for i := 0; i < len(args)-1; i += 2 {
+			if k, ok := args[i].(string); ok {
+				m[k] = args[i+1]
+			}
+		}
+		return m
+	}
+
+	w.trace = ao.NewTraceWithOptions("aws_lambda_go",
+		ao.SpanOptions{
+			ContextOptions: ao.ContextOptions{
+				MdStr: mdStr,
+				CB:    cb,
+			},
+		},
+	)
+	w.mdStr = mdStr
+	txnName := strings.TrimLeft(method+"."+lambdacontext.FunctionName, ".")
+	w.trace.SetTransactionName(txnName)
+	w.trace.SetMethod(method)
+	return ao.NewContext(ctx, w.trace)
+}
+
+type typedError struct {
+	typ string
+	err error
+}
+
+func (w *traceWrapper) After(result interface{}, err *typedError, endArgs ...interface{}) interface{} {
+	statusCode := 0
+	errType := ""
+	errClass := ""
+	errMsg := ""
+
+	if err != nil {
+		errType = err.typ
+		errClass = "error"
+		errMsg = err.err.Error()
+		statusCode = 500
+	}
+
+	if statusCode == 0 {
+		if gwRsp, ok := result.(events.APIGatewayProxyResponse); ok {
+			statusCode = gwRsp.StatusCode
+		} else if gwRsp, ok := result.(events.APIGatewayV2HTTPResponse); ok {
+			statusCode = gwRsp.StatusCode
+		}
+		if statusCode/100 == 5 {
```

Comment:
what's the reason you divide by 100 and compare against 5 instead of just compare the status code against 500? maybe this is a shorter way than having `statusCode >= 500 && statusCode < 600`?

Reply:
Yes, as the status code may be some other numbers 5xx instead of 500.